logbook-spring-webflux/src/main/java/org/zalando/logbook/spring/webflux/BufferingServerHttpRequest.java

```diff
@@ -30,16 +30,9 @@ final class BufferingServerHttpRequest extends ServerHttpRequestDecorator {
         super(delegate);
         this.serverRequest = serverRequest;
         this.writeHook = writeHook;
-    }
-
-    @Override
-    public HttpMethod getMethod() {
-        HttpMethod method = super.getMethod();
-        // the only way so far to check if request has body or not
-        if (METHODS_WITHOUT_BODY.contains(method)) {
+        if (METHODS_WITHOUT_BODY.contains(super.getMethod()) || !serverRequest.shouldBuffer()) {
```

Comment:
Interesting, so now writeHook will be executed immediately, probably it's fine in case of requests...

Why's `!serverRequest.shouldBuffer()`required?  Removing it doesn't fail any tests.

Reply:
The idea is to log requests as early as possible. Depending on the application, request processing might take some time. Without the `!serverRequest.shouldBuffer()`, POST requests in a `without-body` strategy are logged when `getBody()` is invoked, and this can be delayed

Reply:
Understood, thank you. I merged the PR